platform/mds/mds/src/main/java/org/motechproject/mds/lookup/LookupExecutor.java

```diff
@@ -61,8 +61,10 @@ public Object execute(Map<String, ?> lookupMap, QueryParams queryParams) {
             return MethodUtils.invokeMethod(dataService, lookup.getMethodName(),
                     args.toArray(new Object[args.size()]),
                     argTypes.toArray(new Class[argTypes.size()]));
-        } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+        } catch (NoSuchMethodException | IllegalAccessException e) {
             throw new LookupExecutorException("Unable to execute lookup " + lookup.getLookupName(), e);
+        } catch (InvocationTargetException e) {
+            throw new LookupExecutorException("Unable to execute lookup " + lookup.getLookupName() + ". " + e.getTargetException().getMessage() + ".", e);
```

Comment:
We cannot just retrieve the not localizable message from the underlying exception and display it to the user. We must verify ourselves that the problem is connected to the lookup execution and wrap it into our own, localizable exception, that clearly explains the problem.